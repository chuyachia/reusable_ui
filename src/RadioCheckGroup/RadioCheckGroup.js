/* global console */
import React from "react";
import PropTypes from "prop-types";

const RadioCheckGroup = ({
  children,
  variant,
  className,
  direction,
  onChange,
  selected,
}) => {
  var childrenProps = { variant, direction, groupSelection: selected };
  return (
    <div className={className} onChange={onChange}>
      {children &&
        React.Children.map(children, child =>
          React.cloneElement(child, { ...childrenProps })
        )}
    </div>
  );
};
RadioCheckGroup.defaultProps = {
  onChange: e => {
    console.log(e.target.value);
  },
};
RadioCheckGroup.propTypes = {
  children: PropTypes.node,
  onChange: PropTypes.func,
  variant: PropTypes.string,
  direction: PropTypes.string,
  className: PropTypes.string,
  controlled: PropTypes.bool,
  childrenType: PropTypes.oneOf(["checkbox", "radio"]),
  selected: function(props, propName) {
    if (props["childrenType"] === "checkbox" && !Array.isArray(props[propName]))
      return new Error(
        "Selected should be an array if childrenType is checkbox"
      );
    if (
      props["childrenType"] === "radio" &&
      !(
        typeof props[propName] === "string" || props[propName] instanceof String
      )
    )
      return new Error("Selected should be a string if childrenType is radio");
  },
};

export default RadioCheckGroup;
