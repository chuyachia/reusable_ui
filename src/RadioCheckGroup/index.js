import styled from "@emotion/styled";

import RadioCheckGroup from "./RadioCheckGroup";

export default styled(RadioCheckGroup)`
  display: inline-block;
  vertical-align: middle;
`;
