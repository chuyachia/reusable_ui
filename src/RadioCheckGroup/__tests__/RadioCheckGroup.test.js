import React from "react";

import RadioCheckGroup from "../RadioCheckGroup";

describe("RadioCheckGroup ", () => {
  let wrapper;
  it("should render", () => {
    wrapper = shallow(<RadioCheckGroup />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render its children with properties", () => {
    wrapper = shallow(
      <RadioCheckGroup variant="testvariant" direction="testdirection">
        <span>Item</span>
      </RadioCheckGroup>
    );
    expect(wrapper.find("span").length).toBe(1);
    expect(wrapper.find("span").prop("variant")).toBe("testvariant");
  });
});
