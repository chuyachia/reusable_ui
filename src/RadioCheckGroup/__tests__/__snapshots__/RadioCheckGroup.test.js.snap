// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioCheckGroup  should render 1`] = `<div />`;
